### DISCUSSION TITLE
What's the process for submitting an addon for the official extension?

### DISCUSSION THREAD

**Discussion by the-trumpeter** [ScratchAddons/ScratchAddons#7858] [Discussion]
I know and am sorry, this probably sounds like a stupid question! I'm pretty new to GitHub and the Scratch Addons contribution system, so I want to make sure I have the facts straight:

So I've come up with an epic add-on, I clone the SA repository and start drafting the scripts on my own. Several weeks and one JavaScript murder later, I've got an awesome addon that I wanna add to the official extension on the Chrome Web Store and all that.
What do I do? Where do I submit it? What part of it do I submit?
So, in basic format, what is the process for submitting an addon for consideration?

Again, sorry for asking a stupid question, just wanna make sure I know what I should be doing (otherwise I might and probably will mess it up and annoy everyone even more!!!) Thank you!!!!!! 😁😁

**Comment by scratchusernamemrtbts** [comment]
refer to [CONTRIBUTING.md](https://github.com/ScratchAddons/ScratchAddons/blob/master/.github/CONTRIBUTING.md)

instead of cloning the repo you should fork the repo then push stuff on to your fork
when you're done just open a PR

**Reply by WorldLanguages** [reply]
> So, in basic format, what is the process for submitting an addon for consideration?

Yeah, start by creating a pull request on this repository, @the-trumpeter
If you've never made a pull-request before, it may take some time to understand what's happening

**Reply by the-trumpeter** [reply]
> instead of cloning the repo you should fork the repo then push stuff on to your fork
> when you're done just open a PR

[complete and utter bafflement]

**Comment by DNin01** [comment]
No, it's okay! Everyone starts somewhere.

On GitHub, people can use what are called "pull requests" to propose changes they made to apply to the project.

You'll find more information on our [contributing guidelines](https://github.com/ScratchAddons/ScratchAddons/blob/master/.github/CONTRIBUTING.md) and [developer documentation](https://scratchaddons.com/docs/develop/getting-started/).

Feel free to follow up if you need additional guidance along the way.
